docs/adr/0006-process-for-onboarding-new-tenants.md

```diff
@@ -24,22 +24,17 @@ Onboarding is a relatively simple part of the process where we create a new Tena
-- Enroll a tenant to the service
-- Unenroll a tenant from the service
+Once we have provided a discovery mechanism, we need to define a way in which we can gather the necessary information needed to enroll a tenant to use a service. We are intending to make this to work by defining a common schema through which a service can communicate both the configuration it requires as well as the services upon which it depends. Services can then attach a manifest file containing this information to their Service Tenant via a well known property key, allowing the Management API to obtain the manifest as part of the discovery process.
 
-The idea being that once the Management API has discovered a service, it can request the information that would be required to enroll a tenant to it. This will be a list of required configuration settings, in a known form that allows them to be easily captured and validated before passing them to the enrollment endpoint.
+Since the process of enrollment and unenrollment is standard across tenants, the actual implementation of this can form part of the Management API, driven by the data in the manifests. If we ever encounter a situation where services need to perform non-standard actions as part of tenant enrollment, we can extend the process to provide support a way in which services can be notified of new enrollments - this could be a simple callback URL, or potentially a broadcast-type system using something like Azure Event Grid. Since we don't yet have any services that would need this, we will not attempt to define that mechanism at this time.
 
-The enrollment endpoint for a service will do two things:
-- Firstly, it will attach the relevant configuration to the tenant that's being enrolled.
-- Secondly, if the service that's being enrolled in has dependencies on other services, it will create a new sub-tenant of its own Service Tenant that will be used to access those dependent services. This new subtenant will then be enrolled for the dependent service.
-
-Whilst we did not wish to tightly couple the Marain services to the Management API, this restriction does not apply for services that depend upon other services. As such if a service does have a dependency, the location of that service (i.e. its URI) will be part of the configuration for that service, making it easy for the enrollment endpoint of a service to create a new tenant and enroll it for the dependent service.
+Enrolling a tenant to use a service does two things:
+- Firstly, it will attach the relevant configuration for the service to the tenant that's being enrolled.
+- Secondly, if the service that's being enrolled in has dependencies on other services, it will create a new sub-tenant of the Service Tenant for the service being enrolled that the service will use when accessing dependencies on behalf of the client. This new subtenant will then be enrolled for each of the depended-upon services.
```

Comment:
Do we need to be explicit about the fact that this includes indirect dependencies? (I think the example goes on to cover this, so maybe not.)

Reply:
Made a slight tweak to the wording to make this clearer.